### PR TITLE
[BACKPORT] Reset local metadata of partition upon run of clear-all operation.

### PR DESCRIPTION
**Pull request by ahmetmircik**
cherry-picked backport of https://github.com/hazelcast/hazelcast/pull/13227

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-lab-run

**Comment by blazember** [comment]
run-lab-run

**Comment by blazember** [comment]
run-lab-run

File: hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java [hunk 1]
```diff
@@ -51,69 +48,63 @@
...
+    public LocalUpdatePolicy localUpdatePolicy;
 
-    protected ClientConfig createClientConfig() {
-        ClientConfig clientConfig = new ClientConfig();
-        clientConfig.setProperty("hazelcast.invalidation.max.tolerated.miss.count", "0");
-        clientConfig.setProperty("hazelcast.invalidation.reconciliation.interval.seconds", Integer.toString(RECONCILIATION_INTERVAL_SECONDS));
-        return clientConfig;
+    @Test
+    @Ignore("https://github.com/hazelcast/hazelcast/issues/12548")
+    public void ensure_nearCachedClient_and_member_data_sync_eventually_CACHE_ON_UPDATE() {
```
Comment (Donnerbart) [inline hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java]: See comment in `master` PR :)

File: hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java [hunk 2]
```diff
@@ -250,22 +244,40 @@ private NearCacheRecordStore getNearCacheRecordStore() {
...
-        return nearCacheConfig;
+                .setSize(Integer.MAX_VALUE);
+
+        return super.createNearCacheConfig(inMemoryFormat)
+                .setInvalidateOnChange(true)
+                .setLocalUpdatePolicy(localUpdatePolicy)
+                .setEvictionConfig(evictionConfig);
     }
-}
+}
```
Comment (Donnerbart) [inline hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java]: Missing newline at EOF

File: hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java [hunk 3]
```diff
@@ -160,27 +165,26 @@ public void run() throws Exception {
...
-        clientConfig.setProperty("hazelcast.invalidation.max.tolerated.miss.count", "0");
-        clientConfig.setProperty("hazelcast.invalidation.reconciliation.interval.seconds", Integer.toString(RECONCILIATION_INTERVAL_SECONDS));
-        return clientConfig;
+        return newNearCacheConfig()
+                .setName(mapName)
+                .setInvalidateOnChange(true)
+                .setEvictionConfig(evictionConfig);
     }
-}
+}
```
Comment (Donnerbart) [inline hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java]: Missing newline at EOF

File: hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java]: fixed.

File: hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java (hunk 2 above)
Comment (ahmetmircik) [reply hazelcast-client/src/test/java/com/hazelcast/client/cache/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java]: fixed.

File: hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java (hunk 3 above)
Comment (ahmetmircik) [reply hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/invalidation/InvalidationMemberAddRemoveTest.java]: fixed.

**Pull request merged**